modules/CbTarragonDriver/tarragon/CbTarragonContactorControlSerial.cpp

```diff
@@ -62,10 +62,30 @@ bool CbTarragonContactorControlSerial::switch_state(bool on) {
         }
 
         rv_primary = this->switch_contactor(this->primary, on);
+        if (!rv_primary && this->phase_count == 3) {
+            // switch back the secondary to be safe (but it should not be energized at all)
+            this->switch_contactor(this->secondary, false, false);
+            return false;
+        }
 
         // now the secondary contactor should have also switched
         if (this->phase_count == 3) {
-            rv_secondary = this->secondary.get_feedback_state() == types::cb_board_support::ContactorState::Closed;
+            auto secondary_contactor_state {this->secondary.get_feedback_state()};
+
+            rv_secondary = secondary_contactor_state == types::cb_board_support::ContactorState::Closed;
+
+            // switch_contactor would have raised an error in case we had waited
+            // but since we didn't we have to take care here
+            if (!rv_secondary) {
+                auto actual_contactor_state {on ? types::cb_board_support::ContactorState::Open
+                                                : types::cb_board_support::ContactorState::Closed};
+
+                this->on_error(this->secondary.get_name(), on, actual_contactor_state);
+
+                // switch back both to be on safe side
+                this->switch_contactor(this->primary, false, false);
+                this->switch_contactor(this->secondary, false, false);
```

Comment:
Not sure, if we should better switch off before reporting the error

Reply:
Switching off here is not time-critial: we assume that we are electrically safe when we try to switch on and we also assume that this does not change due to our switch request. Doing the switch-off request _after_ reporting the error allows to have the object still in "desired" state (thought this is not used at the moment in the reporting path).